### ISSUE TITLE
Live markdown for money request description

### ISSUE THREAD

**Issue by roryabraham** [Expensify/App#39192]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1711641837108169

## Action Performed:
1. Press on the global create button
1. Press `Request Money` -> `Manual` -> enter any amount -> choose any participants
1. Press on the description field
1. type markdown text into the description field

_Note:_ There are more permutations of this, the intention with this issue is that it should capture all money request description fields, including:

- 1:1 requests, splits, and send money
- different types of requests such as manual v.s. scan v.s. distance

## Expected Result:
The markdown styling should render in real-time, as it does for the main chat composer.

## Actual Result:
No live markdown formatting.

## Workaround:
n/a

## Platforms:
Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~014fca642f1cb0ae8d</li>
        <li>Upwork Job ID: 1773387024042512384</li>
        <li>Last Price Increase: 2024-03-28</li>
    </ul>
</details>

**Comment by roryabraham** [comment]
accidentally included the upwork tag from the previous issue in a copy/pasta situation. Going to try re-applying `External` to see if we can get new upwork jobs created

**Comment by joekaufmanexpensify** [comment]
Proposals still pending review

**Comment by roryabraham** [comment]
for context - there was some suggestion(s) from SWM folks that perhaps we should wait for live markdown on web to go to prod before working on this.

**Comment by joekaufmanexpensify** [comment]
Cool cool. Can we still identify the proposal we're going to use though?

**Comment by roryabraham** [comment]
Neither proposal is particularly detailed. I think the changes are simple, but what I'd like to see more details on is the testing and rollout plan that we would use to implement them

**Comment by joekaufmanexpensify** [comment]
I'd go ahead with reviewing for now to see if we can find a good proposal

**Comment by joekaufmanexpensify** [comment]
Pending proposal review

**Comment by alitoshmatov** [comment]
@ShridharGoel @GandalfGwaihir Thank you for your proposal, as mentioned before, I also think your proposal lacks a lot of details. You have to consider how to handle styling and logic. As far as of my testings none of them are easily achievable

**Comment by joekaufmanexpensify** [comment]
@alitoshmatov will you have a chance to review today?

**Comment by alitoshmatov** [comment]
> This is already done here: https://github.com/Expensify/App/pull/39519

This looks like  very similar issues

**Comment by alitoshmatov** [comment]
Yes expressed my [concern](https://github.com/Expensify/App/issues/39189#issuecomment-2043091627), let's wait for response from internal engineer: 

> https://github.com/Expensify/App/issues/39192 - We have the same feature request for money request description. Wouldn't it be better to apply live markdown in all descriptions on a single PR?

> If not we would hold our issue so that logic for live markdown on description is finalized and applied and then apply changes on the same manner

**Comment by joekaufmanexpensify** [comment]
Bumped 1:1

**Comment by roryabraham** [comment]
Since we already have a PR to establish the pattern [here](https://github.com/Expensify/App/pull/39519), and that's recognized in @ShridharGoel's proposal, I'm not too concerned about diverging implementations.

Going to assign this to @ShridharGoel - and I recommend that you wait for https://github.com/Expensify/App/pull/39519 to be merged to open your PR.

**Comment by roryabraham** [comment]
> Should we just incorporate this into the existing PR for live markdown with task descriptions https://github.com/Expensify/App/issues/39189#issuecomment-2043091627?

No, we should not. We should ship these PRs separately because, even though the implementation is similar, all the testing and QA can be separate. This issue requires thorough testing of all money request flows already, which is a pretty heavy testing burden. We don't need to pile that on top of an existing PR.

**Comment by joekaufmanexpensify** [comment]
Got it, thanks for confirming!

**Comment by roryabraham** [comment]
https://github.com/Expensify/App/pull/39519 is merged so we should feel free to proceed with the PR now

**Comment by joekaufmanexpensify** [comment]
Great. TY!

**Comment by roryabraham** [comment]
https://github.com/Expensify/App/pull/40224 is up for review

**Comment by joekaufmanexpensify** [comment]
automation was not working, but PR is on production for 7 days. So going to prep for payment today!

**Comment by joekaufmanexpensify** [comment]
This issue was opened prior to April 4th, so we need to issue the following payments: 

- @ShridharGoel - $500 for PR via upwork. 
- @alitoshmatov - $500 for C+ via upwork.

**Comment by joekaufmanexpensify** [comment]
@ShridharGoel please accept our offer in upwork so we can pay you

**Comment by joekaufmanexpensify** [comment]
@alitoshmatov $500 sent and contract ended!

**Comment by joekaufmanexpensify** [comment]
[Bumped in Slack ](https://expensify.slack.com/archives/C01GTK53T8Q/p1714493586736719)

**Comment by joekaufmanexpensify** [comment]
Still pending @ShridharGoel accepting upwork offer

**Comment by joekaufmanexpensify** [comment]
@ShridharGoel $500 sent and contract ended!

**Comment by joekaufmanexpensify** [comment]
Upwork job closed.

**Comment by joekaufmanexpensify** [comment]
All set, TY everyone!
